src/shaders/stdosl.h

```diff
@@ -407,6 +407,35 @@ int hash (point p, float t) BUILTIN;
 
 
 // Displacement functions
+void displace (vector offset) {
+    P += offset;
+    N = normalize (calculatenormal (P));
+}
+void displace (float amp) {
+    displace (amp * N);
+}
+void displace (string space, float amp) {
+    float len = length (transform (space, N));
+    displace (amp * len * N);
+}
+void displace (string space, vector offset) {
+    displace (transform (space, "common", offset));
+}
+
+void bump (vector offset) {
+    N = normalize (calculatenormal (P+offset));
```

Comment:
Isn't it discouraged to change `N` in a surface shader? Most closures take a normal as an argument.

Reply:
I think I would prefer for `bump` and `displace` to return a `normal` and `point` respectively, and let the shader writer manage the assignment to global variables.

Reply:
The idea is that you should call one of these in the "root" shader, which is passed in the accumulated offset or amount.

If you're randomly calling it in mid-graph nodes, perhaps multiple times, you're probably going to do weird things.

Reply:
Also, the idea was, some day (or in some renderers) we might want displace() to have additional side effects. For example, maybe it modifies dPdu, dPdv? Or do other renderer-specific things? I wanted a function that does the "assignment to global variables" part, which may someday grow to have per-renderer trickiness. The "accumulate and return total" part is already trivial.